internal/deploy/parameters.go

```diff
@@ -363,6 +396,89 @@ func (c *ChangeSet) getProcess() (string, error) {
 	return *c.process, nil
 }
 
+func (c *ChangeSet) getVolumeName() (string, error) {
+	if c.volume == nil {
+		return "", newMissingError(FlagVolume)
+	}
+	return *c.volume, nil
+}
+
+func (c *ChangeSet) getVolumes() ([]v1.Volume, error) {
+	volumeName, err := c.getVolumeName()
+	if err != nil {
+		return nil, err
+	}
+	volume := []v1.Volume{{
+		Name: volumeName,
+		VolumeSource: v1.VolumeSource{
+			PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+				ClaimName: volumeName,
+			},
+		},
+	}}
+	return volume, nil
+}
+
+func (c *ChangeSet) getVolumeMounts() ([]v1.VolumeMount, error) {
+	volumeMount := v1.VolumeMount{}
+
+	// set volume mount path if exists
+	if c.volumeMountPath != nil {
+		if c.volume == nil {
+			return nil, fmt.Errorf("%w %s must be used with %s flag",
+				newInvalidUsageError(FlagVolumeMountPath), FlagVolumeMountPath, FlagVolume)
+		}
+		volumeMount.MountPath = *c.volumeMountPath
+	}
+
+	// set volume mount options if exists
+	if c.volumeMountOptions != nil {
+		if c.volume == nil {
+			return nil, fmt.Errorf("%w %s must be used with %s flag",
+				newInvalidUsageError(FlagVolumeMountOptions), FlagVolumeMountOptions, FlagVolume)
+		}
+
+		if value, ok := (*c.volumeMountOptions)["readOnly"]; ok {
+			parsed, err := strconv.ParseBool(value)
+			if err != nil {
+				return nil, fmt.Errorf("%w readOnly must be either true or false",
+					newInvalidUsageError(FlagVolumeMountOptions))
+			}
+			volumeMount.ReadOnly = parsed
+		}
+	}
+
+	// if empty, no mount paths or options were set
+	if volumeMount == (v1.VolumeMount{}) {
```

Comment:
Ha, makes sense, but a little unorthodox.

Reply:
very true, think its okay for this case? I didnt feel the need to check every field within this struct and rather used this short hand (albeit kinda unconventional)